conu/backend/docker/container.py

```diff
@@ -8,49 +8,112 @@
 
 from conu.backend.docker.image import DockerImage
 from conu.apidefs.container import Container
-from conu.utils.core import run_cmd, random_str
+from conu.apidefs.exceptions import ConuException
+from conu.utils.core import run_cmd
+
 
 logger = logging.getLogger(__name__)
 
 
+class DockerRunCommand(object):
+    """
+    helper to execute `docker run` -- users can easily change or override anything
+    """
+
+    def __init__(self, command=None, additional_opts=None):
+        """
+        Build `docker run` command
+
+        :param command: list of str, command to run in the container, examples:
+            - ["ls", "/"]
+            - ["bash", "-c", "ls / | grep bin"]
+        :param additional_opts: list of str, additional options for `docker run`
+        """
+        self.binary = ["docker"]
+        self.global_options = []
+        self.command = ["container", "run"]
+        self.options = [] + (additional_opts or [])
```

Comment:
Why not just `self.options = additional_opts or []`?

Reply:
This is why I love reviews.